homework/06LessonSix.js

```diff
@@ -0,0 +1,173 @@
+'use strict';
+
+var assert = require('assert');
+
+// ****
+// Are you down with OOP?
+// ****
+
+
+// Problem 1:
+// Associative Array Refresh
+// Add properties x: 1, y: 2 to point to make it a point
+// in two dimensional space.
+var point = {
+};
+point['x'] = 1;
+point['y'] = 2;
+
+// Problem 2:
+// Fill in this Point constructor function so that it defines
+// a class of points with properties x and y.
+// In order to do this, you should add parameters x and y
+// as properties to "this".
+// example: this.propName = propValue;
+function Point(x, y) {
+  this.x = x;
+  this.y = y;
+}
+
+// Problem 3:
+// Create a new point using the class constructor Point.
+// Set x to 5, and y to -3
+// This can be done as follows: new Point(someX, someY)
+var anotherPoint = new Point(5, -3);
+
+// ****
+// Concept Checkpoint
+//
+// Write your answer in comments
+//
+// What is the difference between point (an object literal, or associative array)
+//point is the OG/original gangster
```

Comment:
hahaha you got this right! Point is the Class, and var anotherPoint is an instance of that class that inherits all the stuff from Point.